src/jtag/drivers/ftdi_cjtag.c

```diff
@@ -0,0 +1,787 @@
+/**************************************************************************
+*   Copyright (C) 2018 by Martin Hannon                                   *
+*   martin.hannon@ashling.com                                             *
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+*   This program is distributed in the hope that it will be useful,       *
+*   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+*   GNU General Public License for more details.                          *
+*                                                                         *
+*   You should have received a copy of the GNU General Public License     *
+*   along with this program; if not, write to the                         *
+*   Free Software Foundation, Inc.,                                       *
+*   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.           *
+***************************************************************************/
+
+/**
+* @file
+* Handles cJTAG comms to the ARC processor.
```

Comment:
It then, probably, makes sense to rename file to `arc_cjtag` or `arc_ftdi_cjtag` to clearly identify this as ARC-specific.